### PR TITLE
Allow nil values on typed maps

### PR DESCRIPTION
**Pull request by v0idpwn**
Array has a skip_nil? argument, but I don't see why it is needed, so
I didn't add it to map. If wanted, it could be done pretty easily.

### CODE REVIEW

**Comment by v0idpwn** [comment]
ok, just found a problem

File: lib/ecto/type.ex [hunk 1]
```diff
@@ -1216,6 +1216,10 @@ defmodule Ecto.Type do
     :error
   end
 
+  defp map_each([{key, nil} | t], fun, acc) do
+    map_each(t, fun, Map.put(acc, key, nil))
```
Comment (josevalim) [inline lib/ecto/type.ex:1220]: If the type is parameterized, we should still call the type's callback, that's why we have skip_nil?, so we need similar here.

Comment (v0idpwn) [reply lib/ecto/type.ex]: You're absolutely correct. Just fixed it. Happy new year :)

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**